File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    lost = lost.sort((a,b)=> a-b);
    reserve = reserve.sort((a,b)=> a-b);
    
    const list = new Map(reserve.map(e => [e,e])); 
    lost = lost.filter(e=> !list.delete(e))
    lost.forEach((e)=> list.delete(e-1) || list.delete(e+1) ? null : n--);
    
    return n
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    #n = 학생수, lost = 잃어버린 사람, reserve = 여벌 
    _lost = set(lost) - set(reserve)
    _reserve = set(reserve) - set(lost) 
    for r in _reserve:
        front = r - 1 
        back =  r + 1
        if front in _lost:
            _lost.remove(front)
        elif back in _lost:
            _lost.remove(back)
    return n - len(_lost)